nix/hydra/shell.nix

```diff
@@ -108,10 +52,10 @@ let
     name = "hydra-node-cabal-shell";
 
     buildInputs = libs ++ [
-      pkgs.haskell-nix.compiler.${compiler}
-      pkgs.cabal-install
-      pkgs.pkg-config
-    ] ++ buildInputs ++ devInputs;
+      hsPkgs.compiler.${compiler}
+      tools.cabal-install
+      tools.pkg-config
+    ] ++ buildInputs;
```

Comment:
This is a different `cabal-install` than before (Not strictly bad)

And I struggle a bit with `buildInputs` here being just all `tools` (rewritten as list) anyway.

Reply:
What I wanted to try here was to put as much of the data in the flake as possible and have everything else be library calls, rather than data imports in different places. So compiler, tools are defined in sequence top level. I would have just had a buildInputs argument in the shell, but we currently have multiple shells in that function. In any case I think tools is easier for non-nix users to find now.

Reply:
But shouldn't we aim for separating responsibilities as much as possible instead? All of these derivations are _only_ needed to build a dev shell environment and hence are specific to this. Why is better to have their definition in the `flake.nix` than in `shell.nix`? If it is about discoverability, why not having a `tools.nix`?